com.unity.shadergraph/Editor/Drawing/Blackboard/SGBlackboard.cs

```diff
@@ -175,14 +207,195 @@ public void ShowScrollBoundaryRegions()
             }
         }
 
-        public void HideScrollBoundaryRegions()
+        public void OnDragExitedEvent(DragExitedEvent evt)
+        {
+            SetCategoryDragIndicatorVisible(false);
+            HideScrollBoundaryRegions();
+        }
+
+        void HideScrollBoundaryRegions()
         {
             m_BottomResizer.style.visibility = Visibility.Visible;
             m_IsFieldBeingDragged = false;
             m_ScrollBoundaryTop.RemoveFromHierarchy();
             m_ScrollBoundaryBottom.RemoveFromHierarchy();
         }
 
+        int InsertionIndex(Vector2 pos)
+        {
+            int index = -1;
+            VisualElement owner = contentContainer != null ? contentContainer : this;
+            Vector2 localPos = this.ChangeCoordinatesTo(owner, pos);
+
+            if (owner.ContainsPoint(localPos))
+            {
+                index = 0;
+
+                foreach (VisualElement child in Children())
+                {
+                    Rect rect = child.layout;
+
+                    if (localPos.y > (rect.y + rect.height / 2))
+                    {
+                        ++index;
+                    }
+                    else
+                    {
+                        break;
+                    }
+                }
+            }
+
+            if (index == -1 && childCount >= 2)
```

Comment:
A small comment here might be nice. I can guess that this is handling dragging before/after the entire control, but it's not clear to my why `childCount >= 2` is here.

Reply:
In any case, clarified the intent with a comment

Reply:
I would expect those to both work if the childCount is 1 as both are valid locations. Could that be an edge case worth looking into?